File: app/src/main/java/se/gustavkarlsson/aurora_notifier/android/util/UserFriendlyException.java
package se.gustavkarlsson.aurora_notifier.android.util;

public class UserFriendlyException extends RuntimeException {
	private final int stringResourceId;

	public UserFriendlyException(int stringResourceId, Throwable cause) {
		super(cause);
		this.stringResourceId = stringResourceId;
	}

	public UserFriendlyException(int stringResourceId, String message, Throwable cause) {
		super(message, cause);
		this.stringResourceId = stringResourceId;
	}

	public UserFriendlyException(int stringResourceId, String message) {
		super(message);
		this.stringResourceId = stringResourceId;
	}

	public UserFriendlyException(int stringResourceId) {
		this.stringResourceId = stringResourceId;
	}

	public int getStringResourceId() {
		return stringResourceId;
	}
}

File: app/src/main/java/se/gustavkarlsson/skylight/android/util/UserFriendlyException.kt
package se.gustavkarlsson.skylight.android.util

class UserFriendlyException : RuntimeException {
    val stringResourceId: Int

    constructor(stringResourceId: Int, cause: Throwable) : super(cause) {
        this.stringResourceId = stringResourceId
    }

    constructor(stringResourceId: Int, message: String, cause: Throwable) : super(message, cause) {
        this.stringResourceId = stringResourceId
    }

    constructor(stringResourceId: Int, message: String) : super(message) {
        this.stringResourceId = stringResourceId
    }

    constructor(stringResourceId: Int) {
        this.stringResourceId = stringResourceId
    }
}
